optaplanner-examples/src/main/java/org/optaplanner/examples/investment/solver/move/factory/InvestmentBiQuantityTransferMoveIteratorFactory.java

```diff
@@ -107,17 +106,23 @@ public boolean hasNext() {
             } else if (secondTo == secondFrom) {
                 secondTo = allocationList.get(allocationListSize - 2);
             }
-            long firstTransferMillis = RandomUtils.nextLong(workingRandom, firstFrom.getQuantityMillis()) + 1L;
+            long firstTransferMillis = nextLong(workingRandom, firstFrom.getQuantityMillis()) + 1L;
             if (firstFrom == secondFrom && firstFrom.getQuantityMillis() == firstTransferMillis) {
                 // secondTransferMillis must never do a nextLong(0L) which would throw an IllegalArgumentException
                 firstTransferMillis--;
             }
-            long secondTransferMillis = RandomUtils.nextLong(workingRandom, secondFrom.getQuantityMillis()
+            long secondTransferMillis = nextLong(workingRandom, secondFrom.getQuantityMillis()
                     - (firstFrom == secondFrom ? firstTransferMillis : 0L)) + 1L;
             return CompositeMove.buildMove(new InvestmentQuantityTransferMove(firstFrom, firstTo, firstTransferMillis),
                     new InvestmentQuantityTransferMove(secondFrom, secondTo, secondTransferMillis));
         }
 
+        private static long nextLong(Random random, long n) {
+            return random.longs(0, n)
+                    .findAny()
+                    .orElseThrow();
```

Comment:
(important)
This is very different from the original implementation. Are we sure that this is still correct? Uniformly distributed etc?

Original code:
{code}
    public static long nextLong(Random random, long n) {
        // This code is based on java.util.Random#nextInt(int)'s javadoc.
        if (n <= 0L) {
            throw new IllegalArgumentException("n must be positive");
        }
        if (n < Integer.MAX_VALUE) {
            return random.nextInt((int) n);
        }

        long bits;
        long val;
        do {
            bits = (random.nextLong() << 1) >>> 1;
            val = bits % n;
        } while (bits - val + (n - 1L) < 0L);
        return val;
    }
{code}